### PR TITLE
Add AMD (x86_64) and ARM (aarch64) on Heroku-24

### PR DESCRIPTION
**Pull request by schneems**
AMD/x86_64 is the architecture used for all prior base images. Heroku-24 (Ubuntu 24.04) base image is provided with support for ARM/aarch64 (think m1 Mac or graviton AWS server) support. This PR does several things:

- Introduces support for heroku-24 base image
- Build both arm64 and amd64 architecture binaries

### CODE REVIEW

File: README.md
Comment (colincasey) [inline README.md:13]:
```suggestion
Logic lives in the `build.rb` script at the root of this project. It will call `./configure` and `make` with the corresponding inputs. This file is copied into a docker image when `$ bin/build_ruby` is called. See `dockerfiles/Dockerfile.heroku-24` for an example.
```

File: lib/build_script.rb
Comment (colincasey) [inline lib/build_script.rb:28]:
```suggestion
  architecture:, 
  io: $stdout,
```

Comment (schneems) [reply lib/build_script.rb:28]: Weird that standardrb didn't have an opinion about that

File: bin/print_changelog
Comment (colincasey) [inline bin/print_changelog:25]: Did you miss converting this ruby interpolation to bash?
```suggestion
For more information on [Ruby $VERSION, you can view the release announcement](https://www.ruby-lang.org/en/news/).
```

File: dockerfiles/Dockerfile.heroku-24
Comment (colincasey) [inline dockerfiles/Dockerfile.heroku-24:10]: Why is rust installed here?

File: Dockerfile
Comment (colincasey) [inline Dockerfile:1]: What does this `Dockerfile` do?  It seems like a duplicate of `dockerfiles/Dockerfile.heroku-24`. Was it accidentally added by executing the `bin/activate_docker` script?

File: dockerfiles/Dockerfile.heroku-24
Comment (schneems) [reply dockerfiles/Dockerfile.heroku-24:10]: Rust is needed to support the `--yjit` flag https://shopify.engineering/ruby-yjit-is-production-ready

File: Dockerfile
Comment (schneems) [reply Dockerfile:1]: Removed and added it to the .gitignore

**Review by colincasey — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (colincasey) [inline .github/workflows/ci.yml:62]: @schneems You should have a look at what @dzuelke does with [Job Summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/) the PHP automation. It would save you from having to open up the individual step output to get at this content.

Comment (schneems) [reply .github/workflows/ci.yml:62]: That's neat! I implemented it, but the information didn't make it's way to the summary and ended up being hidden from that job's output as well. https://github.com/heroku/docker-heroku-ruby-builder/actions/runs/8527165876/job/23357986389?pr=38 I'll port that over to a new PR to play with instead of blocking this one

**Pull request merged**
